src/Dialect/ONNX/Rewrite.td

```diff
@@ -431,6 +439,18 @@ def CastEliminationPattern : Pat<
 //  (ONNXCastOp (ONNXCastOp $arg, $_), $type),
 //  (ONNXCastOp $arg, $type)>;
 
+//===----------------------------------------------------------------------===//
+// Canonicalization for ONNXCastLikeOp
+//===----------------------------------------------------------------------===//
+
+def CastLikeEliminationPattern:  Pat<
+  // Remove a CastLike pattern if the first input tensor and second input tensor are the same. 
+  (ONNXCastLikeOp $val, $castTo, $saturate),
+  // Remove the castlike.
+  (replaceWithValue $val),
+  // Check that the first input tensor and second input tensor are the same
+  [(HaveSameElementType $val, $castTo), (HaveSameShape $val, $castTo)]>;
```

Comment:
Do you think only `HaveSameElementType` is enough? `HaveSameShape` looks redundant since we use the type of the target only.

Reply:
@tungld The `Equal` did not work I guess because it is designed to compare to ints and not tensors... do you have another suggestion?  Maybe I can try to create another rule?

Reply:
Your `HaveSameElementType` looks enough. Have you tried it only? without `HaveSameShape`,

Reply:
Let me do so now :)

Reply:
It works just fine!